src/index.ts

```diff
@@ -54,28 +56,75 @@ interface ModuleDecoratorOptions {
   only?: boolean;
 }
 
-function qunitSuiteDecorator(
-  target: any,
+interface Constructor {
+  new (...args: any[]): any;
+}
+
+interface TestClassConstructor<I extends TestClass> {
+  new (hooks?: NestedHooks): I;
+}
+
+class TestClass {
+  before?: () => any;
+  after?: () => any;
+  beforeEach?: () => any;
+  afterEach?: () => any;
+}
+
+/**
+ * Recursively walk up a prototype chain, gathering all
+ * properties along the way.
+ *
+ * @param clazz A factory for test suite classes
+ */
+function getClassProperties<C extends Constructor>(clazz: C) {
+  let ret: Partial<InstanceType<C>> = {};
+
+  function methods(obj: Partial<InstanceType<C>>) {
+    if (obj) {
+      let ps = Object.getOwnPropertyNames(obj);
+
+      ps.forEach(p => {
+        if (BANNED_PROTO_PROPS.indexOf(p) >= 0) return;
+        ret[p] = obj[p];
+      });
+
+      methods(Object.getPrototypeOf(obj));
+    }
+  }
+
+  methods(clazz.prototype);
+
+  return ret;
+}
+
+function qunitSuiteDecorator<
+  I extends TestClass,
+  Target extends TestClassConstructor<I>
+>(
+  target: Target,
   name: string,
   options: ModuleDecoratorOptions,
   meta: { [k: string]: any } = {},
   hooks?: Hooks,
   nested?: (hooks: NestedHooks) => void
 ) {
-  let fn: (name: string, cb: (this: any, hooks: NestedHooks) => void) => void = QUnit.module;
+  let fn: (name: string, cb: (this: any, hooks: NestedHooks) => void) => void =
+    QUnit.module;
   if (options.skip) fn = (QUnit.module as any).skip;
   else if (options.only) fn = (QUnit.module as any).only;
   let normalizedName: string = name
     ? name
     : target
-      ? target.name
-      : `Unnamed QUnit Module ${Math.round(1e6 + Math.random() * 1e6).toString(
-          16
-        )}`;
+    ? target.name
+    : `Unnamed QUnit Module ${Math.round(1e6 + Math.random() * 1e6).toString(
+        16
+      )}`;
   let returned: any = fn(normalizedName, function(this: any, hks: NestedHooks) {
     if (nested) nested(hks);
     let instance = new target(hks);
-    Object.assign(this, instance);
+    const r = getClassProperties(target);
+    Object.assign(this, instance, r);
```

Comment:
This is and `getClassProperties` are the meaningfull changes. A deeper explanation is here https://github.com/mike-north/qunit-decorators/issues/35#issuecomment-436856259

but the TL;DR is -- QUnit does some very interesting stuff with `this` in test functions, and I don't want to throw that out in favor of the way `this` typically works within a `class` -- doing so would risk

* breaking compatibility with QUnit plugins
* confusing QUnit users who are used to the standard behavior
* breaking compatibility with standard setups, such as Ember's QUnit tests (pre [RFC-232](https://github.com/emberjs/rfcs/blob/master/text/0232-simplify-qunit-testing-api.md))

Instead, I'm trying to create the union of both worlds -- anything users would be able to access from within the context of a method, and anything they'd expect to be on the testing context should be available at runtime. Unfortunately, to create this "bridged world", the above shenanegans are required